src/dcap_provider.cpp

```diff
@@ -777,19 +795,27 @@ static quote3_error_t get_collateral(
         const auto curl_operation = curl_easy::create(url, request_body);
         curl_operation->perform();
         response_body = curl_operation->get_body();
-        auto get_header_operation =
-            get_unescape_header(*curl_operation, header_name, &issuer_chain);
+        auto get_issuer_chain_operation =
+            get_unescape_header(*curl_operation, issuer_chain_header, &issuer_chain);
+
+        std::string cache_control;
+        auto get_cache_header_operation =
+            get_unescape_header(*curl_operation, headers::CACHE_CONTROL, &cache_control);

```

Comment:
Why are you getting the cache control header when retval is not success? Shouldn't it be inside the "if" lower down?

Reply:
Thanks - fixed